src/analysis/analysis-service.js

```diff
@@ -80,72 +93,44 @@ AnalysisService.prototype.createSourceAnalysisForLayer = function (layerId, laye
 };
 
 /**
- * Update an analysis node
+ * Update an analysis node from a JSON definition
  */
 AnalysisService.prototype.updateAnalysis = function (analysisDefinition) {
   analysisDefinition = _.clone(analysisDefinition);
   var analysis = this.findNodeById(analysisDefinition.id);
 
-  if (analysis) {
-    var analysisAttrs = this._createAnalysisAttributesFromAnalysisDefinition(analysisDefinition);
-    analysis.set(analysisAttrs);
+  if (!analysis) {
+    throw new Error('The analysis ' + analysisDefinition.id + ' does not exist');
   }
 
+  var analysisAttrs = this._getAnalysisAttributesFromAnalysisDefinition(analysisDefinition, this.updateAnalysis.bind(this));
```

Comment:
I don't understand why need to pass `this.updateAnalysis.bind(this)` (and `this.createAnalysis.bind(this)`) to `_getAnalysisAttributesFromAnalysisDefinition`... Couldn't we just invoke `this._analyse(...)` or `this.createAnalysis` inside of `_getAnalysisAttributesFromAnalysisDefinition`?

Reply:
This is to use the same method `_getAnalysisAttributesFromAnalysisDefinition` instead of writing the code three times. It can be done in different ways